### PR TITLE
Add dorebin and docleanup optional arguments to icepack_step_ridge

### LINKED ISSUE #478: ridge_ice no longer public
Hi all, some of us in the SIS2 community have been calling Icepack's ridge_ice from SIS2. I saw that it became private as of Icepack 1.3.4 and started working on updating how we call Icepack's ridging routines. The public icepack_step_ridge calls both ridge_ice and cleanup_itd. I'm not opposed to the idea of cleaning up the itd in a mass and energy conserving fashion, but something you are doing doesn't play well with SIS2.

* It flat out doesn't work when we use a process mask_table
* Without a mask_table it makes the coupled ice-ocean model unbelievably slow, slower than the simple lack of the mask_table can explain.

Can I ask that ridge_ice be made public again?

### PR DESCRIPTION
**Pull request by apcraig**
## PR checklist
- [X] Short (1 sentence) summary of your PR: 
    Add dorebin and docleanup optional arguments to icepack_step_ridge
- [X] Developer(s): 
    kshedstrom, apcraig
- [X] Suggest PR reviewers from list in the column to the right.
- [X] Please copy the PR test results link or provide a summary of testing completed below.
    Tested Icepack and CICE on Derecho without the optional arguments, with the optional arguments set to true and with the optional arguments set to false.  Icepack remained bit-for-bit in all cases (probably too few gridcells), CICE changed answers when the optional arguments were set to false.  Otherwise results are bit-for-bit, but will do more testing before merging.  Ran full test suites on Derecho with intel and gnu for Icepack and CICE just before merging, results look fine.
- How much do the PR code changes differ from the unmodified code? 
    - [X] bit for bit
    - [ ] different at roundoff level
    - [ ] more substantial 
- Does this PR create or have dependencies on CICE or any other models?
    - [ ] Yes
    - [X] No
- Does this PR add any new test cases?
    - [ ] Yes
    - [X] No
- Is the documentation being updated? ("Documentation" includes information on the wiki or in the .rst files from doc/source/, which are used to create the online technical docs at https://readthedocs.org/projects/cice-consortium-cice/.)
    - [ ] Yes
    - [X] No, does the documentation need to be updated at a later time?
        - [ ] Yes
        - [X] No 
- [X] Please document the changes in detail, including _why_ the changes are made.  This will become part of the PR commit log.

Add dorebin and docleanup optional arguments to icepack_step_ridge to support NOT calling cleanup_itd and NOT calling rebin in the ridge_ice.
Closes #478

Rename limit_aice_in argument in ridge_ice to limit_aice.  This argument is not used in Icepack or CICE at the moment.

Based on https://github.com/ESMG/Icepack/tree/optional_cleanup

Update interface document

### CODE REVIEW

**Comment by eclare108213** [comment]
> Do we need new namelist (in Icepack and CICE) to allow users to turn the features on and off or will this just be hardcoded in the calls when needed?

Since we generally would not recommend these flags be set to false, I do not think they should be exposed in namelist.

**Comment by eclare108213** [comment]
Since it's pertinent, please also remove the reference to a boundary call mentioned [here](https://github.com/CICE-Consortium/Icepack/issues/478#issuecomment-1822093194).  It's around line 742 in ice_itd.F90.  I looked in the old svn repository -- that comment is ANCIENT (more than 20 years).

**Comment by kshedstrom** [comment]
I don't understand it, but I can reproduce my old answers if I add yet a third optional argument to icepack_step_ridge. If the third one is false, I call ridge_ice without the opening and closing arguments, forcing it to call ridge_prep. I thought I was doing all the same computations on the SIS2 side, but something is still different. If chinook ever comes back, I'll have a better shot at debugging it.

**Comment by kshedstrom** [comment]
I figured it out - it's all in the order of operations. I approve this PR.

**Comment by apcraig** [comment]
@kshedstrom Great to hear you figured out what's going on.  So do you need both of these optional arguments to make things work?  Do you understand why?

**Comment by kshedstrom** [comment]
No, I only need the one, but I can get three different answers by having both. I haven't talked to the GFDL folks about this and maybe down the road we'll be able to pick which answer we want, but so far we've only tested it in long runs without any itd_cleanup at all.

File: columnphysics/icepack_mechred.F90 [hunk 1]
```diff
@@ -1841,6 +1851,17 @@ subroutine icepack_step_ridge(dt,           ndtd,          &
...
+      if (present(docleanup)) then
+         ldocleanup = docleanup
+      else
+         ldocleanup = .true.
+      endif
+
+      if (present(dorebin)) then
+         ldorebin = dorebin
+      else
+         ldorebin = .true.
```
Comment (eclare108213) [inline columnphysics/icepack_mechred.F90:1863]: `ldorebin` does not appear to be used in this subroutine

**Comment by kshedstrom** [comment]
It is passed through to cleanup_itd.

**Review by eclare108213 — APPROVED** [review]
Ah, now I see `ldorebin` in the `cleanup_itd` argument list.  Thanks for the clarification. I'm still a bit hesitant to approve these changes without understanding what it is about the driver that requires them.  I'm worried about nonphysical results that might not be noticed or accounted for.  But I will approve, since these changes seem important for the GFDL model, and because these options are buried in the code where others are less likely to thoughtlessly experiment with them. If they turn out to be helpful for other groups, then we can expose them in namelist.  I highly recommend the GFDL model's developers look closely at how their model handles ice thicknesses that lie outside of their thickness bins, especially considering @kshedstrom's comment that 
> I only need the one, but I can get three different answers by having both.

**Comment by apcraig** [comment]
@kshedstrom @eclare108213 I am a little concerned that we don't fully understand why we need these changes and whether the science results are valid.  I'm willing to merge this mostly because the feature is pretty well hidden, although the new arguments will appear in our Icepack interface documentation.  @kshedstrom please confirm again that you want these changes as currently implemented in the PR to be merged and that you will continue to validate and report back any results to us.  Thanks.

**Comment by kshedstrom** [comment]
Yes, this is indeed what I want, at least for now. In the MOM6 world having the ability to reproduce old answers can be important and we need one flag for that. I will let you know about the rest as I learn more.

**Comment by apcraig** [comment]
I ran full test suites on Derecho with intel and gnu for Icepack and CICE, tests look good.

**Pull request merged**